sky/skylet/log_lib.py

```diff
@@ -440,6 +447,14 @@ def tail_logs(job_id: Optional[int],
         with open(log_path, 'r', newline='', encoding='utf-8') as log_file:
             # Using `_follow` instead of `tail -f` to streaming the whole
             # log and creating a new process for tail.
+            if tail > 0:
+                lines = collections.deque(log_file.readlines(), maxlen=tail)
+                for line in lines:
+                    print(line, end='')
+                # Flush the last n lines
+                print(end='', flush=True)
+            # Now, the cursor is at the end of the last lines
+            # if tail > 0
             for line in _follow_job_logs(log_file,
                                          job_id=job_id,
                                          start_streaming_at=start_stream_at):
```

Comment:
Similarly, if we have the `start_steam_at` in lines above, this function will never start yield lines, as it never hits `start_stream_at`